### PR TITLE
Fixed the issue [#1824] - resource deletion issue

### PR DESCRIPTION
**Pull request by hyi**
@pkdash Can you code review this small PR when you have a chance? I believe it is a regression issue that used to work but does not work now. I have also added relevant unit tests to cover this specific scenario in which an empty resource is created first in hydroshare zone and then files from federated user zone is added into this resource in hydroshare zone. The added unit tests have passed in my local test environment. It'd be great if this PR can be merged to make into the next deploy if not earlier as a hot fix. Thanks.

### CODE REVIEW

File: hs_core/models.py [hunk 1]
```diff
@@ -1465,10 +1465,18 @@ def delete(self, using=None):
                 if fl.fed_resource_file_name_or_path.find(self.short_id) >= 0:
```
Comment (pkdash) [inline hs_core/models.py]: I am wondering if the following one liner could replace the ```if ... elif.. else```:
```istorage.delete('{}/{}'.format(self.root_path, fl.full_path)```

Comment (hyi) [inline hs_core/models.py]: @pkdash No, it won't. self.root_path and self.resource_federation_path are all resource level properties that only consider the resource root path and federation path which is not sufficient and does not apply the file-level granularity here. In this specific case, resource is in hydroshare zone, but the files being added into the resource comes from a federated user zone that was indicated by the fed_resource_file_name_or_path property. This can probably be simplified by clearing up fed_resource_file_name_or_path property after the file is added into the resource that resides in hydroshare zone rather than a federated zone, but I am afraid I could be breaking things up if I do so at this point, especially given @alvacouch is also doing work to simplify resource files at this time. To minimize effort on my part and maximize efficiency between my efforts and @alvacouch 's effort, I think what I did here is the simplest and safest fix to get the issue resolved without causing any potential regression issues. I also added a unit test case to cover this special scenario so that we will run into less potential regression issues during refactoring work.

Comment (hyi) [inline hs_core/models.py]: @pkdash I was thinking of something else when giving the comments above. Now that I looked at the code again, I think you are right, I can simplify the logic using root_path. Will do so and make sure all tests pass before asking your final review again. Thanks.

**Comment by hyi** [comment]
@pkdash I have addressed your review comment and run federation tests in my local development environment to make sure they all passed. So it is ready for your final review. Thanks.

File: hs_core/tests/api/native/test_irods_user_zone_federation.py [hunk 2]
```diff
@@ -230,3 +230,36 @@ def test_resource_operations_in_user_zone(self):
...
+                         msg="Number of content files is not equal to 1")
+
+        file_list = []
+        for f in res.files.all():
+            file_list.append(f.fed_resource_file_name_or_path.split('/')[-1])
+        self.assertTrue(self.file_one in file_list,
+                        msg='file 1 has not been added in the resource in hydroshare zone')
+        # test original file in user test zone still exist after adding it to the resource
+        # since 'copy' is used for fed_copy_or_move when adding the file to the resource
+        self.assertTrue(self.irods_storage.exists(user_path + self.file_one))
```
Comment (pkdash) [inline hs_core/tests/api/native/test_irods_user_zone_federation.py]: Should we also do this check after the resource gets deleted?

Comment (hyi) [inline hs_core/tests/api/native/test_irods_user_zone_federation.py]: @pkdash Although our current code does not mess with user's scratch space after files become resource's content files, it won't hurt to add this check in just in case. I will add this check in after the resource gets deleted then.

**Comment by hyi** [comment]
@pkdash Have addressed your further comment on unit test and ran the test again in my local development environment and all tests have passed. So it is ready for your final review again. Thanks.

**Comment by pkdash** [comment]
+1. Merge after Jenkins checks are successful.
